### PR TITLE
feat: add towncrier changelog github action

### PR DESCRIPTION
**Pull request by ThibaultFy**
Example of successful workflow run on substrafl: 
https://github.com/Substra/substrafl/actions/runs/10177323059/job/28148559322

To be triggered by workflow dispatch on each repoo.
Example of code to add on each repo with towncrier:

```yaml
name: Towncrier changelog

on:
  pull_request:
    branches:
      - main
  workflow_dispatch:
    inputs:
        app_version:
            type: string
            description: 'The version of the app'
            required: true
        branch:
            type: string
            description: 'The branch to update'
            required: true

jobs:
  test-generate-publish:
    uses: substra/substra-gha-workflows/.github/workflows/towncrier-changelog-build.yml@main
    secrets: inherit
    with:
      app_version: ${{ inputs.app_version }}
      repo: { REPO_NAME_TO_ADD }
      branch: ${{ inputs.branch }}
```

part of FL-1685

### CODE REVIEW

File: .github/workflows/towncrier-changelog.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+jobs:
+  create-changelog:
+    name: Create changelog with towncrier
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout ${{ inputs.repo }}
+        uses: actions/checkout@v4
+        with:
+          repository: substra/${{ inputs.repo }}
+          path: ${{ inputs.repo }}
```
Comment (guilhem-barthes) [inline .github/workflows/towncrier-changelog.yml:28]: Not sure what this line is doing, is it checking out a subset?

Comment (ThibaultFy) [reply .github/workflows/towncrier-changelog.yml:28]: It clone the repu into a folder `path`. I think it is useful when we clone several repo but in this case you're right I think we can remove it and remove the `cd ${{ inputs.repo }}` in the `Run Towncrier` step.
I'll do the changes

Comment (ThibaultFy) [reply .github/workflows/towncrier-changelog.yml:28]: Or maybe it isolating the repo in a folder is a good thing ? Maybe we can keep it to ensure that...

Comment (ThibaultFy) [reply .github/workflows/towncrier-changelog.yml:28]: I checked several implementation and we always define the `path`. I think it's a good practice to keep

**Review by guilhem-barthes — APPROVED** [review]

**Pull request merged**